packages/relic/test/headers/new/special/accept_encoding_header_test.dart

```diff
@@ -0,0 +1,331 @@
+          equals([1.0, 1.0, 1.0]),
+        );
+      },
+    );
+
+    test(
+      'when an Accept-Encoding header with invalid quality values is passed then it should throw TestInvalidHeaderException',
+      () async {
+        expect(
+          () async => await getServerRequestHeaders(
+            server: server,
+            headers: {'accept-encoding': 'gzip;q=abc, deflate, br'},
+            strictHeaders: true,
+          ),
+          throwsA(isA<TestInvalidHeaderException>()),
+        );
+      },
+    );
+
+    test(
+      'when an Accept-Encoding header with extra whitespace is passed then it should parse correctly',
+      () async {
+        Headers headers = await getServerRequestHeaders(
+          server: server,
+          headers: {'accept-encoding': ' gzip , deflate , br '},
+          strictHeaders: true,
+        );
+
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.encoding).toList(),
+          equals(['gzip', 'deflate', 'br']),
+        );
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.quality).toList(),
+          equals([1.0, 1.0, 1.0]),
+        );
+      },
+    );
+  });
+
+  group('Given an Accept-Encoding header with the strict flag false', () {
+    late RelicServer server;
+
+    setUp(() async {
+      try {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv6, 0);
+      } on SocketException catch (_) {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv4, 0);
+      }
+    });
+
+    tearDown(() => server.close());
+
+    test(
+      'when a valid Accept-Encoding header is passed then it should parse correctly',
+      () async {
+        Headers headers = await getServerRequestHeaders(
+          server: server,
+          headers: {'accept-encoding': 'gzip, deflate, br'},
+          strictHeaders: false,
+        );
+
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.encoding).toList(),
+          equals(['gzip', 'deflate', 'br']),
+        );
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.quality).toList(),
+          equals([1.0, 1.0, 1.0]),
+        );
+      },
+    );
+
+    test(
+      'when a valid Accept-Encoding header with quality values is passed then it should parse correctly',
+      () async {
+        Headers headers = await getServerRequestHeaders(
+          server: server,
+          headers: {'accept-encoding': 'gzip;q=1.0, deflate;q=0.5, br;q=0.8'},
+          strictHeaders: false,
+        );
+
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.encoding).toList(),
+          equals(['gzip', 'deflate', 'br']),
+        );
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.quality).toList(),
+          equals([1.0, 0.5, 0.8]),
+        );
+      },
+    );
+
+    test(
+      'when an invalid Accept-Encoding header is passed then it should return null',
```

Comment:
Suggestion: assert the value is available on the other place for non-parsable headers.